application.go

```diff
@@ -365,7 +409,40 @@ func (r *Application) AddEnv(name, value string) *Application {
 // the environments of an application that already has environments set (setting env to nil will
 // keep the current value)
 func (r *Application) EmptyEnvs() *Application {
-	r.Env = &map[string]string{}
+	r.Env = &map[string]EnvValue{}
+
+	return r
+}
+
+// AddSecret adds a secret declaration
+//		name:	the name of the variable
+//		source:	the source ID of the secret
+func (r *Application) AddSecret(name, source string) *Application {
```

Comment:
Question: Does each secret have to go along with an environment secret, possibly sharing certain elements? From the examples in the doc, it looks like the secret value in the environment variable has to match the key in the `secrets` map.

The reason why I'm asking: If they always need to correspond, should we populate both environment variable secrets and `secrets` by this method?